### ISSUE TITLE
`single_target_shortest_path_length` on main returns dict, not iterator

### ISSUE THREAD

**Issue by eriknw** [networkx/networkx#7315]
There seem to be inconsistencies introduced in #6584 and #7161. For example, for `single_target_shortest_path_length`, the warning says that it will return dict (not iterator) starting in version 3.5, but it returns a dict _today_.

**Comment by eriknw** [comment]
The warning message is:
```
single_target_shortest_path_length will return a dict instead of
an iterator in version 3.5
```
but the return statement is:
```python
return dict(_single_shortest_path_length(adj, nextlevel, cutoff))
```

**Comment by rossbar** [comment]
For context: in the latest released version (3.2.1) it still returns a generator:

```python
In [1]: import networkx as nx

In [2]: nx.__version__
Out[2]: '3.2.1'

In [3]: G = nx.path_graph(10)

In [4]: nx.single_target_shortest_path_length(G, 0, 9)
Out[4]: <generator object _single_shortest_path_length at 0x771b69589d80>
```

so this inconsistency is only on the development branch and should be rectified before the next release!
